facebook/delphiFacebook/R/binary.R

```diff
@@ -210,22 +210,43 @@ compute_binary_response <- function(response, weight, sample_size)
 #' @return Updated data frame.
 #' @importFrom dplyr mutate
 jeffreys_binary <- function(df) {
-  return(mutate(df,
-                val = jeffreys_percentage(.data$val, .data$sample_size),
-                se = binary_se(.data$val, .data$sample_size)))
+  jeffreys_temp <- jeffreys_multinomial_factory(2)
+  return(jeffreys_temp(df))
 }
 
-#' Adjust a percentage estimate to use the Jeffreys method.
+#' Generate function that applies Jeffreys correction to multinomial estimates.
 #'
-#' Takes a previously estimated percentage (calculated with num_yes / total *
+#' @param k Number of groups.
+#' 
+#' @return Function to apply multinomial Jeffreys correction.
+#' @importFrom dplyr mutate
+jeffreys_multinomial_factory <- function(k) {
+  # Apply a Jeffreys correction to multinomial estimates and their standard errors.
+  #
+  # Param df: Data frame
+  # Returns: Updated data frame.
+  jeffreys_multinomial <- function(df) {
+    return(mutate(df,
+                  val = jeffreys_percentage(.data$val, .data$sample_size, k),
+                  se = binary_se(.data$val, .data$sample_size)))
+  }
+  
+  return(jeffreys_multinomial)
+}
+
+#' Adjust a multionmial percentage estimate to use the Jeffreys method.
+#'
+#' Takes a previously estimated percentage (calculated with num_group1 / total *
 #' 100) and replaces it with the Jeffreys version, where one pseudo-observation
-#' with 50% yes is inserted.
+#' with 1/k * 100% group1 is inserted.
```

Comment:
`group1` is confusing here. I might write something like "where one pseudo-observation with 1/k mass in each group is inserted".